lib/core/src/restore.rs

```diff
@@ -0,0 +1,1172 @@
+                .lbtc_lockup_script_history
+                .iter()
+                .find(|&tx| tx_map.incoming_tx_map.contains_key::<Txid>(&tx.txid))
+                .cloned();
+
+            let (btc_server_lockup_tx_id, btc_claim_tx_id) = match history
+                .btc_claim_script_history
+                .len()
+            {
+                // Only lockup tx available
+                1 => (Some(history.btc_claim_script_history[0].clone()), None),
+
+                2 => {
+                    let first_tx = history.btc_claim_script_txs[0].clone();
+                    let first_tx_id = history.btc_claim_script_history[0].clone();
+                    let second_tx_id = history.btc_claim_script_history[1].clone();
+
+                    let btc_lockup_script = send_chain_swap_immutable_db_by_swap_id
+                        .get(&swap_id)
+                        .map(|imm| imm.claim_script.clone())
+                        .ok_or_else(|| {
+                            anyhow!("BTC claim script not found for Onchain Send Swap {swap_id}")
+                        })?;
+
+                    // We check the full tx, to determine if this is the BTC lockup tx
+                    let is_first_tx_lockup_tx = first_tx
+                        .output
+                        .iter()
+                        .any(|out| matches!(&out.script_pubkey, x if x == &btc_lockup_script));
+
+                    match is_first_tx_lockup_tx {
+                        true => (Some(first_tx_id), Some(second_tx_id)),
+                        false => (Some(second_tx_id), Some(first_tx_id)),
+                    }
+                }
+                n => {
+                    error!("BTC script history with unexpected length {n} found while recovering data for Chain Send Swap {swap_id}");
+                    (None, None)
+                }
+            };
+
+            res.insert(
+                swap_id,
+                RecoveredOnchainDataChainSend {
+                    lbtc_user_lockup_tx_id,
+                    lbtc_refund_tx_id,
+                    btc_server_lockup_tx_id,
+                    btc_claim_tx_id,
+                },
+            );
+        }
+
+        Ok(res)
+    }
+
+    /// Reconstruct Chain Receive Swap tx IDs from the onchain data and the immutable DB data
+    async fn recover_receive_chain_swap_tx_ids(
+        &self,
+        tx_map: &TxMap,
+        chain_receive_histories_by_swap_id: HashMap<String, ReceiveChainSwapHistory>,
+        receive_chain_swap_immutable_db_by_swap_id: &HashMap<String, ReceiveChainSwapImmutableData>,
+    ) -> Result<HashMap<String, RecoveredOnchainDataChainReceive>> {
+        let mut res: HashMap<String, RecoveredOnchainDataChainReceive> = HashMap::new();
+        for (swap_id, history) in chain_receive_histories_by_swap_id {
+            info!("[Recover Chain Receive] Checking swap {swap_id}");
+
+            let (lbtc_server_lockup_tx_id, lbtc_server_claim_tx_id) = match history
```

Comment:
It is the server lockup and the user claim, isn't it? Shouldn't we rename `lbtc_server_claim_tx_id` to `lbtc_user_claim_tx_id`

Reply:
As discussed elsewhere, they are both "server txs", as they both act on the swapper-locked funds.